src/MultiFormats.Tests/VarintTest.cs

```diff
@@ -0,0 +1,168 @@
+            CollectionAssert.AreEqual(x, Varint.Encode(long.MaxValue));
+            Assert.AreEqual(long.MaxValue, Varint.DecodeInt64(x));
+        }
+
+        [TestMethod]
+        public void Encode_Negative() { ExceptionAssert.Throws<NotSupportedException>(() => Varint.Encode(-1)); }
+
+        [TestMethod]
+        public void TooBig_Int32()
+        {
+            var bytes = Varint.Encode((long) int.MaxValue + 1);
+            ExceptionAssert.Throws<InvalidDataException>(() => Varint.DecodeInt32(bytes));
+        }
+
+        [TestMethod]
+        public void TooBig_Int64()
+        {
+            var bytes = "ffffffffffffffffff7f".ToHexBuffer();
+            ExceptionAssert.Throws<InvalidDataException>(() => Varint.DecodeInt64(bytes));
+        }
+
+        [TestMethod]
+        public void Unterminated()
+        {
+            var bytes = "ff".ToHexBuffer();
+            ExceptionAssert.Throws<InvalidDataException>(() => Varint.DecodeInt64(bytes));
+        }
+
+        [TestMethod]
+        public void Empty()
+        {
+            var bytes = new byte[0];
+            ExceptionAssert.Throws<EndOfStreamException>(() => Varint.DecodeInt64(bytes));
+        }
+
+        [TestMethod]
+        public async Task WriteAsync()
+        {
+            await using (var ms = new MemoryStream())
+            {
+                await ms.WriteVarintAsync(long.MaxValue);
+                ms.Position = 0;
+                Assert.AreEqual(long.MaxValue, ms.ReadVarint64());
+            }
+        }
+
+        [TestMethod]
+        public void WriteAsync_Negative()
+        {
+            var ms = new MemoryStream();
+            ExceptionAssert.Throws<Exception>(() => ms.WriteVarintAsync(-1).Wait());
+        }
+
+        [TestMethod]
+        public void WriteAsync_Cancel()
+        {
+            var ms = new MemoryStream();
+            var cs = new CancellationTokenSource();
+            cs.Cancel();
+            ExceptionAssert.Throws<TaskCanceledException>(() => ms.WriteVarintAsync(0, cs.Token).Wait());
+        }
+
+        [TestMethod]
+        public async Task ReadAsync()
+        {
+            await using (var ms = new MemoryStream("ffffffffffffffff7f".ToHexBuffer()))
+            {
+                var v = await ms.ReadVarint64Async();
+                Assert.AreEqual(long.MaxValue, v);
+            }
+        }
+
+        [TestMethod]
+        public void ReadAsync_Cancel()
+        {
+            var ms = new MemoryStream(new byte[]
+            {
+                0
+            });
+            var cs = new CancellationTokenSource();
+            cs.Cancel();
+            ExceptionAssert.Throws<TaskCanceledException>(() => ms.ReadVarint32Async(cs.Token).Wait());
+        }
+
+        [TestMethod]
+        public void Example()
+        {
+            for (long v = 1; v <= 0xFFFFFFFL; v = v << 4)
+            {
+                Console.Write($"| {v} (0x{v.ToString("x")}) ");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this logging statement.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)